src/main/java/com/blackducksoftware/integration/alert/channel/hipchat/HipChatChannel.java

```diff
@@ -126,25 +131,79 @@ public String testGlobalConfig(final HipChatGlobalConfigEntity entity) throws In
+            throws IntegrationException {
         if (config.getRoomId() == null) {
             throw new IntegrationException("Room ID missing");
         } else {
-            final Collection<ProjectData> projectDataCollection = digestModel.getProjectDataCollection();
-            final String htmlMessage = createHtmlMessage(projectDataCollection);
-            final String jsonString = getJsonString(htmlMessage, AlertConstants.ALERT_APPLICATION_NAME, config.getNotify(), config.getColor());
+            if (isChunkedMessageNeeded(event)) {
+                final String eventContent = event.getContent();
+                final int requestCount = calculateChunks(eventContent);
+                final List<Request> requestList = createChunkedRequestList(channelRequestHelper, config, globalConfig, event, requestCount);
+                return requestList;
+            } else {
+                final String contentTitle = String.format("%s -> %s", event.getProvider(), event.getNotificationType());
+                return Arrays.asList(createRequest(channelRequestHelper, config, globalConfig, contentTitle, event.getContent()));
+            }
+        }
+    }
 
-            final String url = getApiUrl(globalConfig) + "/v2/room/" + config.getRoomId().toString() + "/notification";
+    private boolean isChunkedMessageNeeded(final ChannelEvent event) {
+        final String eventContent = event.getContent();
+        if (StringUtils.isNotBlank(eventContent)) {
+            return eventContent.length() > MESSAGE_SIZE_LIMIT;
+        } else {
+            return false;
+        }
+    }
 
-            final Map<String, String> requestHeaders = new HashMap<>();
-            requestHeaders.put("Authorization", "Bearer " + globalConfig.getApiKey());
-            requestHeaders.put("Content-Type", "application/json");
+    private int calculateChunks(final String eventContent) {
+        final int contentLength = eventContent.length();
+        logger.info("Message too large.  Creating chunks..");
+        logger.info("Content length: {}", contentLength);
+        final int additionPage = (contentLength % MESSAGE_SIZE_LIMIT == 0) ? 0 : 1;
+        final int requestCount = (contentLength / MESSAGE_SIZE_LIMIT) + additionPage;
+        logger.info("Number of requests to submit: {}", requestCount);
+        return requestCount;
+    }
 
-            return channelRequestHelper.createPostMessageRequest(url, requestHeaders, jsonString);
+    private List<Request> createChunkedRequestList(final ChannelRequestHelper channelRequestHelper, final HipChatDistributionConfigEntity config, final HipChatGlobalConfigEntity globalConfig, final ChannelEvent event,
+            final int requestCount) throws IntegrationException {
+        final List<Request> requestList = new ArrayList<>(requestCount);
+        final String eventContent = event.getContent();
+        for (int index = 0; index < requestCount; index++) {
```

Comment:
Couldn't this be a while loop? so that we dont need to calculate the chunks before hand. You can just keep going until the end > eventContent.length()